### ISSUE TITLE
Content of exception_messages string pointer arrays is empty in 18-interrupts

### ISSUE THREAD

**Issue by JayceCao** [cfenollosa/os-tutorial#87]
When learning `18-interrupts`, content of `exception_messages` in `cpu/isr.c` is empty. All items are `0x0`. It seems that `ld` don't replace the string constants with real address.
I do this tutorial on `Gentoo` with `gcc (Gentoo 7.3.0-r3 p1.4) 7.3.0`, `GNU ld (Gentoo 2.30 p5) 2.30.0`. 
`CFLAGS` is `-m32 -fno-pie -g`. There would be errors if I don't use `-fno-pie`.

**Comment by EthanGYoung** [comment]
I had this same error and fixed it by loading more sectors from disk at boot time. I originally was loading 8 into dh, but I changed it to 15 and it worked.

```
load_kernel:
  mov bx, MSG_LOAD_KERNEL ; Print message to say loading kernel
  call print_string
  call print_nl

  mov bx, KERNEL_OFFSET   ; Read from disk and store in 0x1000
  mov dh, 15  
  mov dl, [BOOT_DRIVE]  ; From boot disk
  call disk_load
  ret
```

**Comment by Menotdan** [comment]
> I had this same error and fixed it by loading more sectors from disk at boot time. I originally was loading 8 into dh, but I changed it to 15 and it worked.
> 
> ```
> load_kernel:
>   mov bx, MSG_LOAD_KERNEL ; Print message to say loading kernel
>   call print_string
>   call print_nl
> 
>   mov bx, KERNEL_OFFSET   ; Read from disk and store in 0x1000
>   mov dh, 15  
>   mov dl, [BOOT_DRIVE]  ; From boot disk
>   call disk_load
>   ret
> ```

Yeah, when the kernel gets too big it doesn't load the whole kernel so parts of it are missing and so it bugs out and sometimes crashes. I ran out of space on my extension to the kernel so I had to make it as large as it could be, but then I ran out of space so now im moving to grub for more kernel space

**Comment by jlagarespo** [comment]
Use grub guys, not only will your operating system be more universal, but it will also not have to worry about booting up the computer

**Comment by Menotdan** [comment]
Just make sure if you post your code on github to include a tarball that you built grub from so you comply with the license for grub

**Comment by jlagarespo** [comment]
??
Does the grub license say anything about including a tarball of it?
Can't you just install grub and use it?

**Comment by Menotdan** [comment]
Well on the OS dev barebones tutorial wiki page, it says you have to have a tarball of the grub you built in your source somewhere if you are going to share it
